test/vsphere/e2e.sh

```diff
@@ -0,0 +1,159 @@
+  export AWS_SECRET_ACCESS_KEY=$SOPS_AWS_SECRET_ACCESS_KEY
+
+  for filename in "$CERTS"/*.enc; do
+    # save the decrypted certs in the same folder w/o the ext .enc
+    sops --config "$PWD"/test/vsphere/.sops.yaml -d \
+      --input-type binary --output-type binary "$filename" >"${filename%.*}"
+  done
+
+  # clean a WARNING in openvpn
+  # WARNING: file '/openvpn/certs/user.key' is group or others accessible
+  chmod 400 "$CERTS"/user.key
+}
+
+dump_logs() {
+  mkdir -p "$ARTIFACTS"/snapshot
+  platform-cli snapshot --output-dir "$ARTIFACTS"/snapshot
+}
+
+on_exit() {
+  # dump the logs into the ARTIFACTS directory
+  dump_logs
+
+  # remove the cluster
+  platform-cli cleanup
+
+  # kill the VPN
+  docker kill vpn
+
+  # clean certs dir - remove all except enc
+  rm "$CERTS"/{*.crt,*.key}
+}
+
+wait_for_vpn() {
+  while [ -n "$(ip addr show tun0 2>&1 >/dev/null)" ]; do
+    sleep 0.1
+  done
+}
+
+generate_cluster_id() {
+  local prefix
+
+  prefix=$(tr </dev/urandom -cd 'a-f0-9' | head -c 5)
+  echo "e2e-${prefix}"
+}
+
+# These are used inside e2e.yml
+PLATFORM_CLUSTER_ID=$(generate_cluster_id)
+export PLATFORM_CLUSTER_ID
+export PLATFORM_CA=$PWD/.certs/ca.crt.pem
+export PLATFORM_PRIVATE_KEY=$PWD/.certs/ca.key.pem
+export PLATFORM_CA_CEK=foobar
+export PLATFORM_OPTIONS_FLAGS="-e name=${PLATFORM_CLUSTER_ID} -e domain=${PLATFORM_CLUSTER_ID}.lab.flanksource.com"
+
+export PLATFORM_CONFIG=$PWD/test/vsphere/e2e.yml
+
+# These variables are required to configure access to vSphere, they also work with govc
+export GOVC_FQDN="vcenter.lab.flanksource.com"
+export GOVC_DATACENTER="lab"
+export GOVC_CLUSTER="cluster"
+export GOVC_NETWORK="VM Network"
+export GOVC_PASS="${GOVC_PASS}"
+export GOVC_USER="${GOVC_USER}"
+export GOVC_DATASTORE="datastore1"
+export GOVC_INSECURE=1
+export GOVC_URL="$GOVC_USER:$GOVC_PASS@$GOVC_FQDN"
+
+trap on_exit EXIT
+
+install_platformcli
+
+install_sops
+
+# Decrypt certs encrypted with SOPS and KMS Key from https://github.com/flanksource/vsphere-lab/blob/master/.sops.yaml
+decrypt_vpn_certs
+
+# Run the vpn client in container
+docker run --rm -d --name vpn -v "${PWD}/test/vsphere/openvpn/:/openvpn/" \
+  -w "/openvpn/" --cap-add=NET_ADMIN --net=host --device=/dev/net/tun \
+  gcr.io/cluster-api-provider-vsphere/extra/openvpn:latest
+
+echo "nameserver 10.255.0.2" | sudo tee -a /run/resolvconf/resolv.conf >/dev/null
+
+## Wait for VPN
+wait_for_vpn
+
+# Tail the vpn logs
+docker logs vpn
+
+dig +short $GOVC_FQDN
+
+# Generate Ingress certs
+platform-cli ca generate --name ingress-ca \
+  --cert-path "$PLATFORM_CA" --private-key-path "$PLATFORM_PRIVATE_KEY" \
+  --password "$PLATFORM_CA_CEK" --expiry 1
+
+# Create the cluster using the config from PLATFORM_CONFIG
+# shellcheck disable=SC2086
+platform-cli provision vsphere-cluster $PLATFORM_OPTIONS_FLAGS
+
+# Build the base platform configuration
+# shellcheck disable=SC2086
+
+platform-cli build all $PLATFORM_OPTIONS_FLAGS
```

Comment:
It fails here with the message:

```
INFO[0000] Installing gomplate (v3.5.0) -> https://github.com/hairyhenderson/gomplate/releases/download/v3.5.0/gomplate_linux-amd64 

ERRO[0000] Error deploying base rbac: applySpecs: failed to template manifests: /home/circleci/project/.bin/gomplate -f "%s" -o "%s" -c ".=%s" failed with: exit status 127, stderr: bash: /home/circleci/project/.bin/gomplate: No such file or directory
```

Despite `gotemplate` is installed. I believe it is instellated in the wrong destination and it could not be found here.

Reply:
I think the problem may be that the `.bin` directory does not exist

Reply:
Then this would fails https://github.com/flanksource/platform-cli/pull/139/files#diff-e3dfb84483f9f9b5536b270d1c76180cR33

Reply:
Just checked the path situation and the only obvious thing is that `gomplate` is not installed under `/home/circleci/project/.bin` but somewhere else while the folder `.bin` is actually where it is supposed to be.

Hard to debug here where `gomplate` is installed because the installation  is in the middle of a command.

Reply:
The only obvious thing it comes to my mind is that the file is downloaded as `gomplate_linux-amd64` and stored there with the file. when CLI does the lookup with `gomplate` it fails.

Reply:
@teodor-pripoae By chance do you know the error here? I could not get why this fails.